Nodes/SerialIO.c

```diff
@@ -306,8 +309,14 @@ char gettekn(void) {
       UpdateInactive();
     }
     if((signals & serreadsig) && CheckIO((struct IORequest *)serreadreq)) {
-      ch = sergettkn();
-      ConvChrsToAmiga(&ch,1,Servermem->inne[nodnr].chrset);
+      int conv;
+      serbuf[seridx] = sergettkn();
+      conv = ConvMBChrsToAmiga(&ch,serbuf,seridx+1,
+                               Servermem->inne[nodnr].chrset);
+      if (conv < 0) {
+        ch = 0;
```

Comment:
Is setting it to 0 really needed? It must have been 0 when entering the loop iteration. I guess it could be something different if there was something coming from CON as well. But there should probably be a "break" at the end of that if block.